<launch>
	<arg name="robot_ns" default=""/>
	<arg name="ground_truth" default="false"/>
	<arg name="location" default="table"/>

	<rosparam file="$(find sd_localization)/config/$(arg robot_ns)_localization.yaml" command="load"/>
	
	<node name="$(arg robot_ns)_map_server" pkg="map_server" type="map_server" args="$(find sd_localization)/config/amcl_$(arg location)_map.yaml"/>

	<node pkg="imu_filter_madgwick" type="imu_filter_node" name="$(arg robot_ns)_imu_filter_madgwick" if="$(eval arg('robot_ns') == 'r1')"/>

	<node pkg="robot_localization" type="ukf_localization_node" name="$(arg robot_ns)_localization_odom" clear_params="true" output="screen">
	</node>

	<node pkg="robot_localization" type="ukf_localization_node" name="$(arg robot_ns)_localization_map" clear_params="true" output="screen">
		<remap from="odometry/filtered" to="odom_filtered_map" />
	</node>

	<node pkg="amcl" type="amcl" name="$(arg robot_ns)_amcl" output="screen">
		<remap from="scan" to="laser/scan_filtered"/>
		
		<!-- The following line remap map to r*/map -->
		<remap from="map" to="map"/>
	</node>

	<node pkg="sd_localization" type="sd_localization_snap_map" name="$(arg robot_ns)_snap_map">
		<remap from="laser_scan" to="laser/scan_filtered"/>
	</node>
</launch>